app/authRedirect.js

```diff
@@ -0,0 +1,66 @@
+// Create the main myMSALObj instance
+// configuration parameters are located at authConfig.js
+const myMSALObj = new msal.PublicClientApplication(msalConfig);
+
+let accessToken;
+
+// Register Callbacks for Redirect flow
+myMSALObj.handleRedirectCallback(authRedirectCallBack);
+
+function authRedirectCallBack(error, response) {
+    if (error) {
+        console.error(error);
+    } else {
+        if (myMSALObj.getAccount()) {
+            console.log('id_token acquired at: ' + new Date().toString());
+            showWelcomeMessage(myMSALObj.getAccount());
+            getTokenRedirect(loginRequest);
+        } else if (response.tokenType === "Bearer") {
+            console.log('access_token acquired at: ' + new Date().toString());
+        } else {
+            console.log("token type is:" + response.tokenType);
+        }
+    }
+}
+
+// Redirect: once login is successful and redirects with tokens, call Graph API
+if (myMSALObj.getAccount()) {
+    showWelcomeMessage(myMSALObj.getAccount());
+}
+
+function signIn() {
+    myMSALObj.loginRedirect(loginRequest);
+}
+
+function signOut() {
+    myMSALObj.logout();
+}
+
+// This function can be removed if you do not need to support IE
+function getTokenRedirect(request) {
+    return myMSALObj.acquireTokenSilent(request)
+        .then((response) => {
+            console.log(response);
+            if (response.accessToken) {
+                console.log('access_token acquired at: ' + new Date().toString());
+                accessToken = response.accessToken;
+
+                callMSGraph(graphConfig.graphMeEndpoint, response.accessToken, updateUI);
+                profileButton.style.display = 'none';
+                mailButton.style.display = 'initial';
+            }
+        })
+        .catch(error => {
```

Comment:
maybe console.error(error) here as well?

Reply:
didn't want to make it an error log since it isn't a blocking error. Will change to warn.